benchmarks/python/test_many_pointwise_ops.py

```diff
@@ -0,0 +1,64 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024-present NVIDIA CORPORATION & AFFILIATES.
+# All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause
+import pytest
+from nvfuser import FusionDefinition, DataType
+from nvfuser.pytorch_utils import torch_dtype_to_nvfuser_dtype
+from .core import run_benchmark, clear_cuda_cache
+import torch
+from .global_params import PROMOTE_DTYPES
+from functools import partial
+
+
+def pointwise_ops_fusion(fd: FusionDefinition, dtype: DataType, num_iters: int):
+    x = fd.define_tensor(shape=[-1], contiguity=[True], dtype=dtype, is_cpu=False)
+    y = fd.define_tensor(shape=[-1], contiguity=[True], dtype=dtype, is_cpu=False)
+
+    if dtype in PROMOTE_DTYPES:
+        x = fd.ops.cast(x, dtype=DataType.Float)
+        y = fd.ops.cast(y, dtype=DataType.Float)
+
+    a = fd.ops.add(x, y)
+    for _ in range(num_iters):
+        x = fd.ops.cos(a)
+        y = fd.ops.sin(a)
+        a = fd.ops.add(x, y)
+
+    if dtype in PROMOTE_DTYPES:
+        a = fd.ops.cast(a, dtype=dtype)
+    fd.add_output(a)
+
+
+# NOTE: num_iters restricted due to issue #1234.
+@pytest.mark.parametrize("num_iters", [2, 4, 8, 16])
+def test_pointwise_ops_benchmark(
```

Comment:
I wonder if there is a way to automatically modify the test name if we detect that `device="host"`. Barring that, we should just take care that the test name reflects that this is compile time benchmarking.
```suggestion
def test_pointwise_ops_compile_time_benchmark(
```